### PR TITLE
[estuary] optionally show date/duration if recordings sort mode is size or duration

### PR DESCRIPTION
**Pull request by phunkyfish**
## Description

When adding support for sorting by recording sizes the date field was removed from the right panel.

This PR allows for them when using different the sorting modes of duration/size. When size is selected both date and duration are shown as they can be seen nowhere else. If Duration is selected only data is how.

## Motivation and Context

@ksooo pointed out the omission.

## How Has This Been Tested?

Using iptvsimple and vuplus on MacOSX

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: addons/skin.estuary/xml/Includes_PVR.xml [hunk 1]
```diff
@@ -213,7 +213,7 @@
 					<top>60</top>
 					<height>200</height>
 					<wrapmultiline>true</wrapmultiline>
-					<label>$INFO[ListItem.TimerType,[COLOR grey]$LOCALIZE[803]:[/COLOR] ,[CR]]$INFO[ListItem.Duration,[COLOR grey]$LOCALIZE[180]:[/COLOR] ,[CR]]$VAR[ExpirationDateTimeLabel,[CR]]$VAR[FlagLabel]</label>
+					<label>$INFO[ListItem.TimerType,[COLOR grey]$LOCALIZE[803]:[/COLOR] ,[CR]]$VAR[DateDurationLabel,[CR]]$VAR[ExpirationDateTimeLabel,[CR]]$VAR[FlagLabel]</label>x
```
Comment (ronie) [inline addons/skin.estuary/xml/Includes_PVR.xml:216]: stray `x` at the end of the line

don't think there's a need to include a `[CR]` in the $VAR?
it's already included in the individual infolabels.

likely `$VAR[ExpirationDateTimeLabel,[CR]]` doesn't need it either, could you please double-check?

**Review by ronie — CHANGES_REQUESTED** [review]

File: addons/skin.estuary/xml/Includes_PVR.xml (hunk 1 above)
Comment (phunkyfish) [reply addons/skin.estuary/xml/Includes_PVR.xml:216]: Not sure where the `x` came from!

If I remove the CRs nothing displays. No date/duration/progress bar. It's all blank.

Comment (phunkyfish) [reply addons/skin.estuary/xml/Includes_PVR.xml]: So I just removed the X

Comment (ronie) [reply addons/skin.estuary/xml/Includes_PVR.xml]: > If I remove the CRs nothing displays.

that shouldn't happen... please give this a shot:

`<label>$INFO[ListItem.TimerType,[COLOR grey]$LOCALIZE[803]:[/COLOR] ,[CR]]$VAR[DateDurationLabel]$VAR[ExpirationDateTimeLabel]$VAR[FlagLabel]</label>`

Comment (phunkyfish) [reply addons/skin.estuary/xml/Includes_PVR.xml]: Sorry, you are right. I got mixed up in square brackets!

PR updated ;)

Comment (phunkyfish) [reply addons/skin.estuary/xml/Includes_PVR.xml]: Is this ok now @ronie ?

**Comment by phunkyfish** [comment]
@ronie is this ok now?

**Comment by ronie** [comment]
it's a bit hard for me to test as i do not have a proper pvr setup.
i have to get by using the pvr.demo addon, but that seriously lacks functionality.

as we now (potentially) display these 5 infolabels:
- type
- date
- duration
- expires
- new flag

**and** a progress-bar in the same area, it could lead to this:

overlap

someone would need to double-check if such a thing could occur in any of the pvr windows:
- channels
- recordings
- timers
- timer rules
- search

**Comment by phunkyfish** [comment]
Good point. That could only happen for recordings with size as the sort method possibly. Everything else would be ok as we dropped date Or duration .

Let me check it tomorrow.

**Comment by phunkyfish** [comment]
Ok, confirmed as only effecting recordings when size is the chosen sort mode. Note that for recordings `Type` is not possible and that the progress bar does not have any values associated with it so it's just the bar.

So what I did was make the duration value conditional on whether a not there is a expiration value available. Sorting by size will be used when a user wants to delete recordings and in this case `duration` is the least important field.

So we should be good now @ronie ;)

**Comment by ronie** [comment]
great, thx for the confirmation!

**Pull request merged**